src/libraries/Common/tests/System/FunctionPointerTests.cs

```diff
@@ -299,6 +332,68 @@ private unsafe class FunctionPointerHolder
 
             public class MyClass { }
             public struct MyStruct { }
+            private class PrivateClass { }
+        }
+
+        /// <summary>
+        /// A mock function pointer type used to test the abstract ContainsGenericParameters implementation.
+        /// </summary>
+        private sealed class MockFunctionPointerType : MockType
+        {
+            private readonly Type _returnType;
+            private readonly Type[] _parameterTypes;
+
+            public MockFunctionPointerType(Type returnType, Type[] parameterTypes)
+            {
+                _returnType = returnType;
+                _parameterTypes = parameterTypes ?? Type.EmptyTypes;
+            }
+
+            protected override bool HasElementTypeImpl() => false;
+            public override bool IsGenericParameter => false;
+            public override bool IsFunctionPointer => true;
+            public override Type GetFunctionPointerReturnType() => _returnType;
+            public override Type[] GetFunctionPointerParameterTypes() => _parameterTypes;
+        }
+
+        /// <summary>
+        /// A mock type that contains generic parameters.
+        /// </summary>
+        private sealed class MockGenericParameterType : MockType
```

Comment:
Delete MockGenericParameterType and MockNonGenericType . Use regular typeof(...) on BCL types like List<T> or string instread.